src/Lecture1.hs

```diff
@@ -108,7 +128,20 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Int
+strSum str =
+    let l = words str
+    in sumStrList l
+
+sumStrList :: [String] -> Int
+sumStrList l =
+    if null l 
+    then 0
+    else    
+        let
+            s = read (head l) :: Int
+        in s + sumStrList(tail l)
```

Comment:
There's a standard function `sum` to find the some of all elements in the list. You can also use function `map` to parse every element of the list. So, when using `words`, `map`, `read` and `sum`, you can implement as a nice one-liner 🙂